EDU-1024: tag Go versioning source nodes (#2279)
temporalio/documentation: docs-src/go/versioning.md

You can simply terminate all of the currently running Workflow Executions and suspend new ones from being created while you deploy the new version of your Workflow code, which does not use `GetVersion()`, and then resume Workflow creation.
However, that is often not the case, and you need to take care of the currently running Workflow Executions, so using `GetVersion()` to update your code is the method to use.

However, if you want your currently running Workflows to proceed based on the current Workflow logic, but you want to ensure new Workflows are running on new logic, you can define your Workflow as a new `WorkflowType`, and change your start path (calls to `StartWorkflow()`) to start the new Workflow type.

## Sanity checking

The Temporal client SDK performs a sanity check to help prevent obvious incompatible changes.

> The sanity check verifies whether a Command made in replay matches the event recorded in history, in the same order. The Command is generated by calling any of the following methods:

Reviewer: ```suggestion
The sanity check verifies whether a Command made in replay matches the event recorded in history, in the same order.
The Command is generated by calling any of the following methods:
```

- workflow.SignalExternalWorkflow()
- workflow.Sleep()

> Adding, removing, or reordering any of the above methods triggers the sanity check and results in a non-deterministic error.

Reviewer: ```suggestion
Adding, removing, or reordering any of the preceding methods triggers the sanity check and results in a non-deterministic error.
```

> The sanity check does not perform a thorough check. For example, it does not check on the Activity's input arguments or the timer duration.
> If the check is enforced on every property, then it becomes too restricted and harder to maintain the Workflow code.
> For example, if you move your Activity code from one package to another package, that changes the `ActivityType`, which technically becomes a different Activity.
> But, we don't want to fail on that change, so we only check the function name part of the `ActivityType`.

Reviewer: ```suggestion
The sanity check does not perform a thorough check.
For example, it does not check on the Activity's input arguments or the Timer duration.
If the check is enforced on every property, it becomes too restrictive and harder to maintain the Workflow code.
For example, if you move your Activity code from one package to another package, that move changes the `ActivityType`, which technically becomes a different Activity.
But we don't want to fail on that change, so we check only the function name part of the `ActivityType`.
```
